Blank Canvas: Add "Coming Soon" Patterns

kjellr opened the pull request:
Closes #3523. This PR begins to add @melchoyce's "Coming Soon" block patterns. 

Todo: 

- [x] Add the rest of the images to the theme, instead of hot linking. 
- [x] Add image credits. 
- [x] Add image alt text.

kjellr committed:
Add initial draft of patterns.

melchoyce commented:
For reference, here's the media sources:
- Restaurant: https://stocksnap.io/photo/nature-food-AUVLL0XR1V
- Band: https://stocksnap.io/photo/people-man-5PEZV88YAV
- Video Background: https://www.pexels.com/video/rock-band-concert-857173/ (note the cc0 license)
- Art Gallery: https://www.metmuseum.org/art/collection/search/436840
- Simple Colorful: https://icons8.com/icon/set/bullhorn/windows
- Startup: https://www.humaaans.com/

melchoyce commented:
Updated patterns:

```
<!-- wp:cover {"url":"https://mywptesting.site/wp-content/uploads/2021/03/StockSnap_AUVLL0XR1V-scaled.jpg","id":2579,"dimRatio":0,"focalPoint":{"x":"1.00","y":"0.51"},"minHeight":100,"minHeightUnit":"vh","align":"full"} -->
<div class="wp-block-cover alignfull" style="min-height:100vh"><img class="wp-block-cover__image-background wp-image-2579" alt="" src="https://mywptesting.site/wp-content/uploads/2021/03/StockSnap_AUVLL0XR1V-scaled.jpg" style="object-position:100% 51%" data-object-fit="cover" data-object-position="100% 51%"/><div class="wp-block-cover__inner-container"><!-- wp:columns {"align":"full"} -->
<div class="wp-block-columns alignfull"><!-- wp:column {"width":"65%"} -->
<div class="wp-block-column" style="flex-basis:65%"></div>
<!-- /wp:column -->

<!-- wp:column {"width":"35%"} -->
<div class="wp-block-column" style="flex-basis:35%"><!-- wp:paragraph -->
<p><strong>OAK &amp; ANCHOR</strong></p>
<!-- /wp:paragraph -->

<!-- wp:spacer {"height":250} -->
<div style="height:250px" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:heading {"style":{"typography":{"fontSize":"64px"}}} -->
<h2 style="font-size:64px"><strong>Coming Soon</strong></h2>
<!-- /wp:heading -->

<!-- wp:paragraph -->
<p>A new restaurant featuring farm-to-table dining.</p>
<!-- /wp:paragraph -->

<!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:column -->
<div class="wp-block-column"><!-- wp:paragraph -->
<p>1234 S Main Street<br>Townsville, MA, 01010</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class="wp-block-column"><!-- wp:paragraph {"style":{"color":{"text":"#ffffff","link":"#ffffff"}}} -->
<p class="has-text-color has-link-color" style="--wp--style--color--link:#ffffff;color:#ffffff"><a href="mailto:info@example.com">info@example.com<br></a>(123) 456-7890</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns -->

<!-- wp:social-links {"iconColor":"background","iconColorValue":"#FFFFFF","className":"is-style-logos-only"} -->
<ul class="wp-block-social-links has-icon-color is-style-logos-only"><!-- wp:social-link {"url":"#","service":"facebook"} /-->

<!-- wp:social-link {"url":"#","service":"yelp"} /-->

<!-- wp:social-link {"url":"#","service":"instagram"} /--></ul>
<!-- /wp:social-links --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:cover -->

<!-- wp:cover {"url":"https://mywptesting.site/wp-content/uploads/2021/03/pattern-links-gradient-scaled.jpg","id":2794,"dimRatio":0,"minHeight":100,"minHeightUnit":"vh","align":"full"} -->
<div class="wp-block-cover alignfull" style="min-height:100vh"><img class="wp-block-cover__image-background wp-image-2794" alt="" src="https://mywptesting.site/wp-content/uploads/2021/03/pattern-links-gradient-scaled.jpg" data-object-fit="cover"/><div class="wp-block-cover__inner-container"><!-- wp:heading {"textAlign":"center","align":"full","style":{"typography":{"fontSize":"20px","lineHeight":"1"}},"textColor":"background"} -->
<h2 class="alignfull has-text-align-center has-background-color has-text-color" style="font-size:20px;line-height:1"><strong><span class="uppercase">COMING SOON</span></strong></h2>
<!-- /wp:heading -->

<!-- wp:heading {"textAlign":"center","level":3,"align":"full","style":{"typography":{"fontSize":"120px","lineHeight":"1"}}} -->
<h3 class="alignfull has-text-align-center" style="font-size:120px;line-height:1"><strong><span class="uppercase">Six Feet Over</span></strong></h3>
<!-- /wp:heading -->

<!-- wp:paragraph {"align":"center"} -->
<p class="has-text-align-center">A new album from Rage Slump coming April 2021. Follow us on BandCamp now to be the first to hear some sneak peeks of the album.</p>
<!-- /wp:paragraph -->

<!-- wp:buttons {"contentJustification":"center"} -->
<div class="wp-block-buttons is-content-justification-center"><!-- wp:button {"textColor":"background","className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link has-background-color has-text-color"><strong>FOLLOW US</strong></a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div></div>
<!-- /wp:cover -->

<!-- wp:cover {"url":"https://mywptesting.site/wp-content/uploads/2021/03/Gradient-bluegreen-slower_low.mp4","id":2808,"dimRatio":0,"overlayColor":"primary","backgroundType":"video","minHeight":100,"minHeightUnit":"vh","align":"full"} -->
<div class="wp-block-cover alignfull has-primary-background-color" style="min-height:100vh"><video class="wp-block-cover__video-background intrinsic-ignore" autoplay muted loop playsinline src="https://mywptesting.site/wp-content/uploads/2021/03/Gradient-bluegreen-slower_low.mp4" data-object-fit="cover"></video><div class="wp-block-cover__inner-container"><!-- wp:columns {"align":"full"} -->
<div class="wp-block-columns alignfull"><!-- wp:column {"width":"75%"} -->
<div class="wp-block-column" style="flex-basis:75%"><!-- wp:heading {"style":{"typography":{"lineHeight":"1","fontSize":"125px"}}} -->
<h2 style="font-size:125px;line-height:1"><strong>The High Fives</strong></h2>
<!-- /wp:heading -->

<!-- wp:social-links {"align":"left"} -->
<ul class="wp-block-social-links alignleft"><!-- wp:social-link {"url":"#","service":"soundcloud"} /-->

<!-- wp:social-link {"url":"#","service":"bandcamp"} /-->

<!-- wp:social-link {"url":"#","service":"facebook"} /-->

<!-- wp:social-link {"url":"#","service":"instagram"} /-->

<!-- wp:social-link {"url":"#","service":"youtube"} /--></ul>
<!-- /wp:social-links --></div>
<!-- /wp:column -->

<!-- wp:column {"width":"25%"} -->
<div class="wp-block-column" style="flex-basis:25%"></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:cover -->

<!-- wp:cover {"customOverlayColor":"#e8ffe6","minHeight":100,"minHeightUnit":"vh","align":"full"} -->
<div class="wp-block-cover alignfull has-background-dim" style="background-color:#e8ffe6;min-height:100vh"><div class="wp-block-cover__inner-container"><!-- wp:image {"align":"center","id":2742,"width":100,"height":100,"sizeSlug":"medium","linkDestination":"none"} -->
<div class="wp-block-image"><figure class="aligncenter size-medium is-resized"><img src="https://mywptesting.site/wp-content/uploads/2021/03/bullhorn-circle-300x300.png" alt="" class="wp-image-2742" width="100" height="100"/></figure></div>
<!-- /wp:image -->

<!-- wp:spacer {"height":5} -->
<div style="height:5px" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:heading {"textAlign":"center","align":"wide","style":{"typography":{"fontSize":"100px","lineHeight":"1.2"}},"textColor":"primary"} -->
<h2 class="alignwide has-text-align-center has-primary-color has-text-color" style="font-size:100px;line-height:1.2"><strong>Coming Soon</strong></h2>
<!-- /wp:heading -->

<!-- wp:paragraph {"align":"center","textColor":"primary"} -->
<p class="has-text-align-center has-primary-color has-text-color">Sign up to hear when my journey begins.</p>
<!-- /wp:paragraph -->

<!-- wp:buttons {"contentJustification":"center"} -->
<div class="wp-block-buttons is-content-justification-center"><!-- wp:button {"borderRadius":50,"backgroundColor":"primary"} -->
<div class="wp-block-button"><a class="wp-block-button__link has-primary-background-color has-background" style="border-radius:50px">Subscribe to my newsletter</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons -->

<!-- wp:social-links {"iconColor":"primary","iconColorValue":"#000000","align":"center","className":"is-style-logos-only"} -->
<ul class="wp-block-social-links aligncenter has-icon-color is-style-logos-only"><!-- wp:social-link {"url":"#","service":"facebook"} /-->

<!-- wp:social-link {"url":"#","service":"twitter"} /-->

<!-- wp:social-link {"url":"#","service":"instagram"} /--></ul>
<!-- /wp:social-links --></div></div>
<!-- /wp:cover -->

<!-- wp:cover {"overlayColor":"background","minHeight":948,"minHeightUnit":"px","align":"full"} -->
<div class="wp-block-cover alignfull has-background-background-color has-background-dim" style="min-height:948px"><div class="wp-block-cover__inner-container"><!-- wp:columns {"align":"full"} -->
<div class="wp-block-columns alignfull"><!-- wp:column -->
<div class="wp-block-column"><!-- wp:heading {"style":{"typography":{"lineHeight":"0.9","fontSize":"120px"}},"textColor":"primary"} -->
<h2 class="has-primary-color has-text-color" style="font-size:120px;line-height:0.9"><strong>Coming Soon</strong></h2>
<!-- /wp:heading -->

<!-- wp:paragraph {"textColor":"primary"} -->
<p class="has-primary-color has-text-color">A new blog by MindStomp. Sign up to be notified when we launch.</p>
<!-- /wp:paragraph -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button {"textColor":"primary","className":"is-style-outline"} -->
<div class="wp-block-button is-style-outline"><a class="wp-block-button__link has-primary-color has-text-color">Sign up</a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class="wp-block-column"></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:cover -->

<!-- wp:media-text {"align":"full","mediaPosition":"right","mediaId":2787,"mediaLink":"https://mywptesting.site/coming-soon-colors/stocksnap_6embfcxu0j-cropped/","mediaType":"image","mediaWidth":54,"imageFill":false} -->
<div class="wp-block-media-text alignfull has-media-on-the-right is-stacked-on-mobile" style="grid-template-columns:auto 54%"><figure class="wp-block-media-text__media"><img src="https://mywptesting.site/wp-content/uploads/2021/03/StockSnap_6EMBFCXU0J-cropped-1024x778.jpg" alt="" class="wp-image-2787 size-full"/></figure><div class="wp-block-media-text__content"><!-- wp:heading {"style":{"color":{"text":"#2b353a"},"typography":{"fontSize":"50px","lineHeight":"1.2"}}} -->
<h2 class="has-text-color" style="color:#2b353a;font-size:50px;line-height:1.2"><strong>A new app for designers</strong></h2>
<!-- /wp:heading -->

<!-- wp:paragraph {"textColor":"primary"} -->
<p class="has-primary-color has-text-color">We're launching soon. Be the first to sign up!</p>
<!-- /wp:paragraph -->

<!-- wp:buttons -->
<div class="wp-block-buttons"><!-- wp:button {"style":{"color":{"background":"#dfcbda","text":"#2b353a"}}} -->
<div class="wp-block-button"><a class="wp-block-button__link has-text-color has-background" style="background-color:#dfcbda;color:#2b353a"><strong><span class="uppercase">Sign up</span></strong></a></div>
<!-- /wp:button --></div>
<!-- /wp:buttons --></div></div>
<!-- /wp:media-text -->

<!-- wp:media-text {"align":"full","mediaId":2719,"mediaLink":"https://mywptesting.site/coming-soon-curated/self-portrait-with-two-pupils-2/","mediaType":"image","imageFill":true,"style":{"color":{"background":"#f3efec"}}} -->
<div class="wp-block-media-text alignfull is-stacked-on-mobile is-image-fill has-background" style="background-color:#f3efec"><figure class="wp-block-media-text__media" style="background-image:url(https://mywptesting.site/wp-content/uploads/2021/03/self-portrait-with-two-pupils-2-871x1024.jpg);background-position:50% 50%"><img src="https://mywptesting.site/wp-content/uploads/2021/03/self-portrait-with-two-pupils-2-871x1024.jpg" alt="" class="wp-image-2719 size-full"/></figure><div class="wp-block-media-text__content"><!-- wp:cover {"customOverlayColor":"#f3efec","minHeight":100,"minHeightUnit":"vh","align":"full"} -->
<div class="wp-block-cover alignfull has-background-dim" style="background-color:#f3efec;min-height:100vh"><div class="wp-block-cover__inner-container"><!-- wp:heading {"style":{"typography":{"fontSize":"64px","lineHeight":"1"}},"textColor":"foreground"} -->
<h2 class="has-foreground-color has-text-color" style="font-size:64px;line-height:1">Opening Soon</h2>
<!-- /wp:heading -->

<!-- wp:paragraph {"style":{"typography":{"lineHeight":"0.5"}},"textColor":"foreground"} -->
<p class="has-foreground-color has-text-color" style="line-height:0.5">A new art gallery in downtown Boston.</p>
<!-- /wp:paragraph -->

<!-- wp:spacer {"height":150} -->
<div style="height:150px" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->

<!-- wp:columns -->
<div class="wp-block-columns"><!-- wp:column -->
<div class="wp-block-column"><!-- wp:paragraph {"textColor":"foreground"} -->
<p class="has-foreground-color has-text-color">1234 S Main Street<br>Boston, MA, 01010</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column -->

<!-- wp:column -->
<div class="wp-block-column"><!-- wp:paragraph {"textColor":"foreground"} -->
<p class="has-foreground-color has-text-color"><a href="mailto:info@example.com">info@example.com<br></a>(123) 456-7890</p>
<!-- /wp:paragraph --></div>
<!-- /wp:column --></div>
<!-- /wp:columns --></div></div>
<!-- /wp:cover --></div></div>
<!-- /wp:media-text -->

<!-- wp:spacer -->
<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
<!-- /wp:spacer -->
```

Preview: https://mywptesting.site/coming-soon-colors/

kjellr committed:
Merge branch 'trunk' into blank-canvas/add/coming-soon-patterns

kjellr committed:
Update patterns

kjellr committed:
Make strings translatable + escaped.

kjellr committed:
Update gradient pattern URL.

kjellr committed:
Add gradient video.

melchoyce commented:
Alt text:

Restaurant: Physalis fruits in a white bowl against a dark background.
Gradient: A blurry gradient blending between shades of purple and blue.
Animated Gradient: A blurry gradient slowly moving between shades of green and blue.
Sign up: A megaphone icon with sharp angles.
App: A close-up of a black smart watch against a pink background.
Art Gallery: "Self-Portrait with Two Pupils" by Marie Gabrielle Capet and Marie Marguerite Carreaux de Rosemond, hung up on a gallery wall.

kjellr committed:
Add new images

kjellr commented:
@melchoyce: I'm not totally sure how to cite the icons8 icon license for their bullhorn graphic. Their license page says that it's "open source" for established open source projects, but requires a link otherwise. Not sure if we qualify for the expanded license since this is a theme, not WordPress itself. 🤔

kjellr committed:
Add image credits.

kjellr committed:
Add alt text for the patterns.

kjellr committed:
Remove alt text from cover block to prevent block errors.

kjellr commented:
I've added the correct images, and all of the image credit ([except for the one noted above](https://github.com/Automattic/themes/pull/3547#issuecomment-813402800)). I've also added the alt text, though I omitted the alt text for the cover blocks, since that's blocked by this bug in Gutenberg: https://github.com/WordPress/gutenberg/issues/30505. 

This could use a review in the meantime. If these look good, should I add them to Blank Canvas blocks in this same PR?

kjellr commented:
Screenshot of all the new patterns: 

![Screen Shot 2021-04-06 at 10 36 42](https://user-images.githubusercontent.com/1202812/113586542-a3100c80-95fb-11eb-815a-1403fbd81cc3.png)

kjellr committed:
Add border color to "Solid Color" pattern form field.

pbking commented:
I think adding them to BCB at the same time would be good.  The patterns LGTM but I'm a terrible judge of these kinds of things so I'm not sure that I would trust me.

melchoyce commented:
> @melchoyce: I'm not totally sure how to cite the icons8 icon license for their bullhorn graphic. Their license page says that it's "open source" for established open source projects, but requires a link otherwise. Not sure if we qualify for the expanded license since this is a theme, not WordPress itself. 🤔

I guess we shouldn't risk it 😕 I'll find a different icon to use.

melchoyce commented:
Maybe we could use this public domain icon instead? https://thenounproject.com/term/bullhorn/89665/

kjellr committed:
Update pattern-notice.png

kjellr committed:
Add license information for the bullhorn icon.

kjellr committed:
Add patterns to Blank Canvas Blocks

kjellr committed:
Add licensing information.

kjellr commented:
I've updated that pattern to include the icon and updated the licensing. I've also migrated these patterns over to Blank Canvas Blocks as well. 

Blank Canvas blocks is missing [the patterns defined in `inc/wpcom.php`](https://github.com/Automattic/themes/blob/a2bed96c957d0d964e40088c8574322aa0f1264f/blank-canvas/inc/wpcom.php#L26-L271). Is that an oversight, @pbking? For the time being I've omitted those for that theme.

pbking committed:
Added 'coming-soon' block pattern category to BCB.  Adjusted namespaces and paths to work in BCB.

pbking committed:
Adjusted 'Coming Soon: Restaurant' pattern to use BCB color palette.

pbking commented:
@kjellr I wasn't even aware of the patterns you linked to there.  As suspect it was probably just an oversight.

pbking commented on blank-canvas/inc/block-patterns.php:
```diff
@@ -497,6 +502,162 @@ function blank_canvas_register_block_patterns() {
+									<!-- /wp:cover --></div></div>
+									<!-- /wp:media-text -->
+
+									<!-- wp:spacer -->
+									<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
+									<!-- /wp:spacer -->',
```
Is this spacer block intentional?
None of the other patterns have that and it seems out of place.

kjellr replied:
Good catch! I've removed these.

pbking commented on blank-canvas-blocks/inc/block-patterns.php:
```diff
@@ -501,6 +501,162 @@ function blank_canvas_register_block_patterns() {
+									<!-- /wp:cover --></div></div>
+									<!-- /wp:media-text -->
+
+									<!-- wp:spacer -->
+									<div style="height:100px" aria-hidden="true" class="wp-block-spacer"></div>
+									<!-- /wp:spacer -->',
```
Is this spacer block intentional?  No other patterns have it and it seems out of place.

pbking commented:
I have checked these against Blank Canvas.  Looks good (except for one note about a spacer block in one pattern).

I have checked these against Blank Canvas Blocks.  There were some namespacing, path and palette issues in there that I cleaned up but otherwise looked good.  Eyes on those changes would be good.

kjellr commented:
> @kjellr I wasn't even aware of the patterns you linked to there. As suspect it was probably just an oversight.

Cool, I'll file a separate issue to track that then. 👍

kjellr committed:
Remove unused spacer blocks.

kjellr committed:
Revise extra-large font sizes for better appearance on small screens.

kjellr committed:
Fix color markup.

kjellr commented:
I double checked your updates @pbking and they're looking great. Thank you! 

I'll go ahead and merge.
